scrapy/contracts/__init__.py

```diff
@@ -41,7 +41,11 @@ def from_spider(self, spider, results):
         requests = []
         for method in self.tested_methods_from_spidercls(type(spider)):
             bound_method = spider.__getattribute__(method)
-            requests.append(self.from_method(bound_method, results))
+            try:
+                requests.append(self.from_method(bound_method, results))
+            except:
```

Comment:
Does it make sense to catch exceptions inheriting from `Exception` instead of this catch-all for BaseException?

I'd expect KeyboardInterrupt or SystemExit to stop contract runner and return non-zero. From your comments any exception raised on this method is silenced and the contract is skipped, that is what you are fixed as I understand, but should we make a shortcut so a SIGTERM cancel collecting contracts and shutdown the process?

Reply:
@dangra that makes sense, fixed it.